docs/xml-sitemaps.md

```diff
@@ -2,70 +2,164 @@
+It is necessary to [verify the site with Google Search Console](https://support.google.com/webmasters/answer/9008080?hl=en) before you can access information about your site's search results performance. It is recommended to use the HTML file upload solution by committing the file to your project's root directory although it is possible to add the meta tag by filling in the [verification code on the Reading Settings page in the admin](admin://options-reading.php).
 
-In order to modify or add extra data to a URL entry in the sitemap XML you can use the `msm_sitemap_entry` filter. Functions attached to this filter can access the current post data using `get_post()` or template tags such as `get_the_permalink()`.
+## Adding Custom Sitemaps
+Sitemaps are provided for built-in content types like pages and author archives out of the box. If you want to add a custom sitemap with additional features such as a video sitemap you can register a custom sitemap provider.
 
-The following example changes the update frequency value and priority for the home page:
+To do so create a custom PHP class that extends the abstract `WP_Sitemaps_Provider` class. Then, you can use the `wp_register_sitemap_provider()` function to register it.
+
+The example below shows a minimal implementation for a custom video sitemap:
 
 ```php
-add_filter( 'msm_sitemap_entry', function ( SimpleXMLElement $url ) : SimpleXMLElement {
-	if ( get_the_permalink() !== get_home_url( '/' ) ) {
-		return $url;
+class Video_Sitemap_Provider extends WP_Sitemaps_Provider {
+
+	/**
+	 * Set the name and object type properties. Required.
+	 */
+	public function __construct() {
+		$this->name = 'videos';
+		$this->object_type = 'post';
 	}
 
-	$url->changefreq = new SimpleXMLElement( '<changefreq>always</changefreq>' );
-	$url->priority = new SimpleXMLElement( '<priority>1.0</priority>' );
+	/**
+	 * Return the list of URLs for the current page.
+	 */
+	public function get_url_list( $page ) : array {
+		$videos = new WP_Query( [
+			'post_type' => 'video',
+			'posts_per_page' => 2000,
+			'fields' => 'ids',
+			'paged' => $page,
+		] );
 
-	return $url;
-} );
-```
+		$urls = [];
 
-You can use this filter to add more detailed information to enrich your sitemaps with [Video](https://support.google.com/webmasters/answer/80471), [Image](https://support.google.com/webmasters/answer/178636) and [News](https://support.google.com/webmasters/answer/74288) data.
+		foreach ( $videos->posts as $video_id ) {
+			$urls[] = get_post_meta( $video_id, 'video_url', true );
+		}
 
-## Adding an XML namespace
+		return $urls;
+	}
 
-Some custom extensions to the standard sitemap format are supported but require the addition of one or more XML namespaces for them to validate. This is achieved using the `msm_sitemap_namespace` filter. The below example adds Google's video sitemap schema.
+	/**
+	 * Return the maximum number of pages to output for this sitemap.
+	 */
+	public function get_max_num_pages() {
```

Comment:
```suggestion
	public function get_max_num_pages() : int {
```

Reply:
I think this would throw a warning or error if it doesn't exactly match the abstract which does not have type hinting.

Reply:
Ah, missed that it's from the parent.